### PR TITLE
Features/settings

### PR DESCRIPTION
**Pull request by vmsavka**

### CODE REVIEW

File: CmrUI/Navigation/CmrContainerVC.swift [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+import UIKit
+
+class CmrContainerVC: UIViewController {
+    
+    private var tab: ContainerTab?
+    private var currentVC: UIViewController?
+    
+    var alert: UIAlertController? // Will implement it later, maybe it will replaced by PXAlertView
+    private lazy var activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .gray) // Will move it to separate class later
```
Comment (mark-flixwagon) [inline CmrUI/Navigation/CmrContainerVC.swift]: @vmsavka Do you have any way to mark those things that you plan on doing later? We usually use TODO and then pass over all the TODO items.

File: CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    let backgroundColor = UIColor(red: 235.0/255.0, green: 235.0/255.0, blue: 235.0/255.0, alpha: 1.0)
+}
+
+protocol NewsFeedVMProtocol {
+    func updateDataSource()
+}
+
+class NewsFeedVM: NSObject {
+    
+        var dataSource = [FeedItem]()
```
Comment (mark-flixwagon) [inline CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift:24]: @vmsavka This is better, but it's not a flexible model accessor - you're assuming an array. The better way would be something like feedItemForRowAtItemIndex or something like this. (Internally, it goes to the array)

File: CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                ,
+                     feedType: FeedType.feedText),
+            FeedItem(image: UIImage(named: Constants().kImgName + "\(5)"), title: "Fifth title", subtitle: "Fifth subtitle", content: "Fifth content", feedType: FeedType.feedImage)
+        ]
+        
+        dataSource.sort(by: sorterTime)
+    }
+    
+    func sorterTime(this: FeedItem, that: FeedItem) -> Bool {
```
Comment (mark-flixwagon) [inline CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift:49]: @vmsavka Yes, but the FeedItems have no time at all yet, and the title is still hard coded as WEDNESDAY xxxxx

File: CmrUI/Helpers/GradientHelper.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+//  Created by Vasyl.Savka on 7/19/18.
+//  Copyright © 2018 Vasyl.Savka. All rights reserved.
+//
+
+import UIKit
+import ObjectiveC
+
+private var GradientLayerKey: UInt8 = 0
+
+class GradientHelper: NSObject {
```
Comment (mark-flixwagon) [inline CmrUI/Helpers/GradientHelper.swift]: @vmsavka If we're receiving a UIView in all calls to this class, then it is a UIView extension. Also, if the gradient is the same, there is no need to associate the object, just use a private var privateGradientLayer: CALayer

File: CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift (hunk 2 above)
Comment (vmsavka) [reply CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift:24]: fixed

File: CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift (hunk 3 above)
Comment (vmsavka) [reply CmrUI/Screens/NewsFeedTab/NewsFeedVM.swift:49]: fixed

File: CmrUI/Helpers/GradientHelper.swift (hunk 4 above)
Comment (vmsavka) [reply CmrUI/Helpers/GradientHelper.swift]: fixed

File: CmrUI/Screens/SettingsTab/SettingsVC.swift [hunk 5]
```diff
@@ -43,8 +43,7 @@ class SettingsVC: UIViewController {
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         
-        GradientHelper.addGradient(onTopOf: gradientView,
-                                   colors: [constants.darkGradientColor,
+        gradientView.addGradient(colors: [constants.darkGradientColor,
```
Comment (mark-flixwagon) [inline CmrUI/Screens/SettingsTab/SettingsVC.swift]: @vmsavka Why call this in viewDidAppear? Can't you do it in viewDidLoad and then count on viewDidLayoutSubviews to update the bounds?

File: CmrUI/Helpers/GradientHelper.swift [hunk 6]
```diff
@@ -29,25 +29,26 @@ class GradientHelper: NSObject {
...
     
-    class func setGradientLayer(_ layer: CALayer?, for view: UIView?) {
-        objc_setAssociatedObject(view ?? UIView(), &GradientLayerKey, layer, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+    func setGradientLayer(_ layer: CALayer?) {
+        objc_setAssociatedObject(self, "🔑", layer, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
     }
     
-    class func updateGradientLayerFrame(on view: UIView?) {
-        let layer: CALayer? = self.gradientLayer(for: view)
+    func updateGradientLayerFrame() {
```
Comment (mark-flixwagon) [inline CmrUI/Helpers/GradientHelper.swift]: @vmsavka When do we actually update the layout? In the app, do we need to adjust for layout changes? The tab bar is static, it's not going to change its size as far as I can tell. Unless I am missing something.

File: CmrUI/Helpers/GradientHelper.swift [hunk 7]
```diff
@@ -29,25 +29,26 @@ class GradientHelper: NSObject {
...
     }
     
-    class func gradientLayer(for view: UIView?) -> CALayer? {
-        if view == nil || view?.layer.sublayers == nil {
+    func gradientLayer() -> CALayer? {
+        if self.layer.sublayers == nil {
             return nil
         }
-        return objc_getAssociatedObject(self, &GradientLayerKey) as? CALayer
+        let ppgog = objc_getAssociatedObject(self, "🔑") as? CALayer
```
Comment (mark-flixwagon) [inline CmrUI/Helpers/GradientHelper.swift]: @vmsavka This line is unnecessary.

File: CmrUI/Helpers/GradientHelper.swift [hunk 8]
```diff
@@ -29,25 +29,26 @@ class GradientHelper: NSObject {
...
+        return objc_getAssociatedObject(self, "🔑") as? CALayer
```
Comment (mark-flixwagon) [inline CmrUI/Helpers/GradientHelper.swift]: @vmsavka 1. Using associated object here is unnecessary. If you want to identify the layer so as not to add it twice, you can use CALayer's 'name' property. 
2. See my remark on viewDidLoad. If you set the gradient layer on view load, you don't need to check if you need to add it or not every time viewDidAppear is called. 
3. Please use English for keys/values, not emoji.

**Comment by mark-flixwagon** [comment]
@vmsavka Please resolve the conflicts and merge to master

**Pull request merged**